### DISCUSSION TITLE
插入length为447的字符串，milvus为什么抛异常提示字符串超长？

### DISCUSSION THREAD

**Discussion by LuckySevenTc** [milvus-io/milvus#25731] [Q&A and General discussion]
我在验证milvus的时候，通过java sdk插入了一条数据，其中一个字段类型为VARCHAR，插入数据的字符串长度为447，但是milvus抛出异常：
R{exception=Exception: the length (1181) of 0th string exceeds max length (1024), status=1, data=null}
我有两个疑问：
1. milvus关于字符串长度的计算是怎么样的，为什么和实际的字符串长度不一致？
2. milvus官方文档中写了VARCHAR类型的长度限制为65536，为什么这里1024就拦截了？

<img width="888" alt="image" src="https://github.com/milvus-io/milvus/assets/44998398/694086dc-4f7d-4ac9-b02d-8f301a1ee2cd">

**Comment by LuckySevenTc** [comment]
我现在知道第二个问题的原因了，在collection创建的时候设置了最大长度为1024。但是第一个问题我还是想请教下大家。

**Comment by xiaofan-luan** [comment]
since 

> 我现在知道第二个问题的原因了，在collection创建的时候设置了最大长度为1024。但是第一个问题我还是想请教下大家。

你好，因为是中文，一个中文会占多个字符。Milvus的长度是按照byte算的

**Comment by Lrcx** [comment]
milvus是按照utf-8下的字节长度进行计算的，比如“啦啦”这两个汉字，字符串长度是2，utf-8编码下的字节为b'\xe5\x95\xa6\xe5\x95\xa6'，长度为6，所以假如设定某个milvus字段的varchar maxlength=4，那么插入啦啦这条数据的时候就会报错。
pymilvus.exceptions.MilvusException: <MilvusException: (code=1100, message=length of varchar field nickname exceeds max length, row number: 0, length: 6, max length: 1: invalid parameter)>
